File: golang/q800/q884.go
package q800

import (
	"strings"
)

func uncommonFromSentences(s1 string, s2 string) []string {
	ws1 := strings.Split(s1, " ")
	ws2 := strings.Split(s2, " ")
	counts := map[string]int{}
	for _, v := range ws1 {
		counts[v]++
	}
	for _, v := range ws2 {
		counts[v]++
	}
	var r []string
	for k, v := range counts {
		if v == 1 {
			r = append(r, k)
		}
	}
	return r
}

File: hs/src/Q800/Q884.hs
module Q800.Q884
(
    uncommonFromSentences
) where

import qualified Data.List as List

uncommonFromSentences :: String -> String -> [String]
uncommonFromSentences s1 s2 = concat . filter (\x -> length x == 1) . List.group . List.sort $ (ws1 ++ ws2)
    where
        ws1 = words s1
        ws2 = words s2
